File: src/ca/mindmagic/game/map/grid/pattern/Shape.java
package ca.mindmagic.game.map.grid.pattern;

import java.awt.Point;

public abstract class Shape {

  private static final int DEFAULT = 40;

  protected int sideLength;

  public Shape(){
    this(DEFAULT);
  }

  public Shape(int sideLength){
    if(sideLength > 0){
      this.sideLength = sideLength;
    }
    else {
      this.sideLength = DEFAULT;
    }
  }

  public abstract int numberOfSides();

  public int lengthOfSides(){
    return sideLength;
  }

  public abstract Point[] getVertices();
}

File: src/ca/mindmagic/game/map/grid/pattern/Shape.kt
package ca.mindmagic.game.map.grid.pattern

import javafx.geometry.Point2D

abstract class Shape @JvmOverloads constructor(var sideLength: Double = DEFAULT) {


    abstract fun getVertices(): Array<Point2D>

    abstract fun numberOfSides(): Int

    open fun lengthOfSides(): Double {
        return sideLength
    }

    companion object {
        private val DEFAULT = 40.0
    }
}
